### DISCUSSION TITLE
Create and chain complex filters

### DISCUSSION THREAD

**Discussion by aeviou** [supabase/supabase#11401] [Questions] [answered]
I'm trying to chain filters together like this:

`SELECT WHERE * ((condition1 AND condition2) OR (condition3 AND condition4) OR ...)`

Specifically, this is what I'm trying to accomplish:
```
DISTANCE_RANGES = [[0, 27], [27, 40], [40, 55], [55, 70], [70, 150], [150, 999]]

CHOSEN_RANGES = []

For all chosen ranges, select the rows that fall within any of the ranges.
```

I can do `.gt(LOWER_BOUND).lt(UPPER_BOUND)` but then it won't work for other ranges. I'm hoping there's some way to do it with `.or()`, such as `distance.gt.${distance.lowerLimit},and(distance.lt.${distance.upperLimit})` but that just applies both conditions to the `or()`.

Is this possible? If so, what's the proper syntax for this?

**Comment by GaryAustin1** [comment, marked as answer]
You can have ands in the .or() filter.
`.or(and(age.gte.7, age.lte.11), and(age.gte.25, age.lte.28))`
https://postgrest.org/en/stable/api.html#logical-operators

**Reply by aeviou** [reply]
Exactly what I was looking for. Thanks!
